ada_2_fhir/zibs2017/payload/zib-payer-2.0.xsl

```diff
@@ -23,173 +23,169 @@ The full text of the license is available at http://www.gnu.org/copyleft/lesser.
-                        <subscriberId>
-                            <xsl:call-template name="string-to-string">
-                                <xsl:with-param name="in" select="."/>
-                            </xsl:call-template>
-                        </subscriberId>
-                    </xsl:for-each>
-                    
-                    <!-- Patient reference -->
-                    <beneficiary>
-                        <xsl:apply-templates select="$adaPatient" mode="doPatientReference-2.1"/>
-                    </beneficiary>
-                    
-                    <xsl:if test="(verzekeraar/verzekering/begin_datum_tijd | insurance_company/insurance/start_date_time)[@value] or (verzekeraar/verzekering/eind_datum_tijd | insurance_company/insurance/end_date_time)[@value]">
-                        <period>
-                            <xsl:for-each select="(verzekeraar/verzekering/begin_datum_tijd | insurance_company/insurance/start_date_time)[@value]">
-                                <start>
-                                    <xsl:attribute name="value">
-                                        <xsl:call-template name="format2FHIRDate">
-                                            <xsl:with-param name="dateTime" select="xs:string(@value)"/>
-                                            <xsl:with-param name="dateT" select="$dateT"/>
-                                        </xsl:call-template>
-                                    </xsl:attribute>
-                                </start>
+        <xsl:variable name="insuranceTypes" select="distinct-values($in/(verzekeraar/verzekering/verzekeringssoort | insurance_company/insurance/insurance_type)/(@code, @originalText)[1])"/>
+        <!--<xsl:variable name="coverages" select="$in[(verzekeraar/verzekering/verzekeringssoort | insurance_company/insurance/insurance_type)/(@code, @originalText)[1] = $insuranceTypes]" as="element()*"/>-->
+        <!--<xsl:for-each-group select="$coverages" group-by="(betaler_persoon, payer_person, (verzekeraar/verzekering/verzekeringssoort | insurance_company/insurance/insurance_type)/(@code, @originalText)[1])[1]">-->
+        <xsl:for-each-group select="$in/betaler_persoon | $in/payer_person | $in/verzekeraar/verzekering | $in/insurance_company/insurance" group-by="(self::betaler_persoon/local-name(), self::payer_person/local-name(), (verzekeringssoort | insurance_type)/(@code, @originalText)[1])[1]">
+            <xsl:variable name="insuranceType" select="current-grouping-key()"/>
+            <xsl:variable name="insuranceCompanies" select="..[verzekering/verzekeringssoort/(@code, @originalText)[1] = $insuranceType] | ..[insurance/insurance_type/(@code, @originalText)[1] = $insuranceType]"/>
+            <xsl:variable name="insurances" select="$insuranceCompanies[verzekering/verzekeringssoort/(@code, @originalText)[1] = $insuranceType] | $insuranceCompanies[insurance/insurance_type/(@code, @originalText)[1] = $insuranceType]"/>
```

Comment:
This should be `$insuranceCompanies/verzekering[verzekeringssoort...] | $insuranceCompanies/insurance[insurance_type...]` (i.e. swap [ and /), which solves the problem of the `.period` not being part of the FHIR instances.

Reply:
Copied this from your branch